### PR TITLE
Fix reliability issue with test case

### LINKED ISSUE #10616: [Bug]: MockKube3ControllersMockTest.testDeploymentController test case is flaky
### Bug Description

Recently I found out that during build the `testDeploymentController` test case failed (but after re-run it was okay). So we should investigate what might be the cause of such a race condition and solve this.
```java
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] io.strimzi.test.mockkube3.MockKube3ControllersMockTest.testDeploymentController
[ERROR]   Run 1: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 2: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 3: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 4: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 5: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
[ERROR]   Run 6: MockKube3ControllersMockTest.testDeploymentController:142 
Expected: is <3>
     but: was null
```
Here is full log from the fail to investigate: [1]

[1] - https://gist.github.com/see-quick/34451c82f12edf865679448355a59e67

### Steps to reproduce

1. Run the build pipeline

### Expected behavior

testDeploymentController test case should not be flaky at all

### Strimzi version

main

### Kubernetes version

1.25.0

### Installation method

_No response_

### Infrastructure

_No response_

### Configuration files and logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MichaelMorrisEst**
### Type of change

- Bugfix (bug in test case)

### Description

Addresses reliability issue with MockKube3ControllersMockTest.testDeploymentController test case by changing the wait condition to wait until after the MockDeploymentController has processed the deployment creation
Closes #10616 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [X] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java [hunk 1]
```diff
@@ -133,8 +133,11 @@ public void testDeploymentController() {
                 .build();
 
         client.apps().deployments().inNamespace(namespace).resource(dep).create();
-
-        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null);
+        
+        TestUtils.waitFor("Wait for deployment to have status", 100L, 10_000L, 
+                () -> client.apps().deployments().inNamespace(namespace).withName(deploymentName).get() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus() != null
+                        && client.apps().deployments().inNamespace(namespace).withName(deploymentName).get().getStatus().getObservedGeneration() != null);
```
Comment (scholzj) [inline mockkube/src/test/java/io/strimzi/test/mockkube3/MockKube3ControllersMockTest.java:140]: I'm not sure what exactly was failing on this. But you are not anymore waiting for it to have non-empty status. So you should update the message.

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR.

**Review by see-quick — APPROVED** [review]
I assume you successfully reproduce this issue locally and after this fix everything works well @MichaelMorrisEst?  Anyway thanks for working on this LGTM 👍.

**Comment by MichaelMorrisEst** [comment]
@see-quick Yes I was able to reproduce locally (intermittently but reasonably often). After applying the fix, despite trying many times I can no longer reproduce, so all looks good

**Pull request merged**